File: clients/java3/client/src/org/openamq/client/handler/ConnectionSecureMethodHandler.java
package org.openamq.client.handler;

import org.openamq.AMQException;
import org.openamq.framing.AMQFrame;
import org.openamq.framing.ConnectionSecureOkBody;
import org.openamq.client.state.AMQStateManager;
import org.openamq.client.state.StateAwareMethodListener;
import org.openamq.client.protocol.AMQMethodEvent;

public class ConnectionSecureMethodHandler implements StateAwareMethodListener
{
    private static final ConnectionSecureMethodHandler _instance = new ConnectionSecureMethodHandler();

    public static ConnectionSecureMethodHandler getInstance()
    {
        return _instance;
    }

    public void methodReceived(AMQStateManager stateManager, AMQMethodEvent evt) throws AMQException
    {
        AMQFrame response = ConnectionSecureOkBody.createAMQFrame(evt.getChannelId(), null);
        evt.getProtocolSession().writeFrame(response);
    }
}

File: dotnet/xmsclient/Client/Handler/ConnectionSecureMethodHandler.cs
using System;
using jpmorgan.mina.common;
using log4net;
using OpenAMQ.Framing;
using OpenAMQ.XMS.Client.Protocol;
using OpenAMQ.XMS.Client.State;

namespace OpenAMQ.XMS.Client.Handler
{
    public class ConnectionSecureMethodHandler : IStateAwareMethodListener
    {
        private static readonly ConnectionSecureMethodHandler _instance = new ConnectionSecureMethodHandler();

        public static ConnectionSecureMethodHandler getInstance()
        {
            return _instance;
        }

        public void MethodReceived(AMQStateManager stateManager, AMQMethodEvent evt)
        {
            AMQFrame response = ConnectionSecureOkBody.CreateAMQFrame(evt.ChannelId, null);
            evt.ProtocolSession.WriteFrame(response);
        }
    }
}
